src/Mixins/Expectation.php

```diff
@@ -817,22 +823,18 @@ public function toMatchSnapshot(string $message = ''): self
             default => InvalidExpectationValue::expected('array|object|string'),
         };
 
-        $testCase = TestSuite::getInstance()->test;
-        assert($testCase instanceof TestCase);
-        $snapshots = TestSuite::getInstance()->snapshots;
-
-        if ($snapshots->has($testCase, $string)) {
-            [$filename, $content] = $snapshots->get($testCase, $string);
+        if ($snapshots->has()) {
+            [$filename, $content] = $snapshots->get();
 
             Assert::assertSame(
                 $content,
                 $string,
                 $message === '' ? "Failed asserting that the string value matches its snapshot ($filename)." : $message
             );
         } else {
-            $filename = $snapshots->save($testCase, $string);
+            $filename = $snapshots->save($string);
 
-            $testCase::markTestIncomplete('Snapshot created at ['.$filename.'].');
+            //$testCase::markTestIncomplete('Snapshot created at ['.$filename.'].');
```

Comment:
This one causes the test to end prematurely, preventing the creation of the second snapshot, I had to comment on it for now, any idea for a workaround?

Reply:
@freekmurze What do you do in your package here?

Reply:
fixed